packages/core/integration-tests/test/javascript.js

```diff
@@ -2656,6 +2656,47 @@ describe('javascript', function () {
     });
   });
 
+  it('should replace __dirname and __filename with path.join calls relative to projectRoot', async function () {
+    let b = await bundle(
+      path.join(__dirname, '/integration/env-node-replacements/index.js'),
+    );
+
+    let dist = await outputFS.readFile(b.getBundles()[0].filePath, 'utf8');
+    assert(
+      dist.includes(
+        'require("path").join(__dirname, "../test/integration/env-node-replacements")',
+      ),
+    );
+    assert(
+      dist.includes(
+        'require("path").join(__dirname, "../test/integration/env-node-replacements", "index.js")',
+      ),
+    );
+    let f = await run(b);
+    let output = f();
+    assert.equal(output.data, 'hello');
+    assert(
+      output.firstDirnameTest.includes(
+        'integration-tests/test/integration/env-node-replacements/data',
+      ),
+    );
+    assert(
+      output.secondDirnameTest.includes(
+        'integration-tests/test/integration/env-node-replacements/other-data',
+      ),
+    );
+    assert(
+      output.firstFilenameTest.includes(
+        'integration-tests/test/integration/env-node-replacements/index.js',
+      ),
+    );
+    assert(
+      output.secondFilenameTest.includes(
+        'integration-tests/test/integration/env-node-replacements/index.js?query-string=test',
+      ),
+    );
```

Comment:
I assume that these fail on Windows because there'll be backslashes

Reply:
Good callout. I'll copy something from our Gatsby repo to deal with this 😅